flake.nix

```diff
@@ -0,0 +1,117 @@
+{
+  description = "vhotplug - Device hot-plugging service for virtual machines";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+
+    flake-parts = {
+      url = "github:hercules-ci/flake-parts";
+      inputs.nixpkgs-lib.follows = "nixpkgs";
+    };
+
+    treefmt-nix = {
+      url = "github:numtide/treefmt-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs =
+    inputs@{ flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.treefmt-nix.flakeModule
+      ];
+
+      systems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+
+      perSystem =
+        {
+          config,
+          self',
+          pkgs,
+          ...
+        }:
+        {
+          # Package definition
+          packages = {
+            default = pkgs.python3Packages.buildPythonApplication {
+              pname = "vhotplug";
+              version = "1.0";
+
+              src = ./.;
+
+              format = "setuptools";
+
+              propagatedBuildInputs = with pkgs.python3Packages; [
+                inotify-simple
+                psutil
+                pyudev
+                qemu-qmp
+              ];
+
+              # Skip tests if they exist
+              doCheck = false;
+
+              meta = with pkgs.lib; {
+                description = "Hot-plugging USB and PCI devices to virtual machines";
+                homepage = "https://github.com/tiiuae/vhotplug";
+                license = licenses.asl20;
+                platforms = platforms.linux;
+                maintainers = [ ];
+              };
+            };
+          };
+
+          # Development shell
+          devShells.default = pkgs.mkShell {
+            inputsFrom = [ self'.packages.default ];
+
+            packages = with pkgs; [
+              # Python development tools
+              python3Packages.pytest
+              python3Packages.mypy
+              python3Packages.pylint
+              # python3Packages.ruff
+
+              # Formatting
+              config.treefmt.build.wrapper
+            ];
+
+            shellHook = ''
+              echo "vhotplug development environment"
+              echo "Run 'treefmt' to format code"
+              echo "Run 'python -m pytest' to run tests"
+            '';
+          };
+
+          # treefmt configuration
+          treefmt = {
+            projectRootFile = "flake.nix";
+
+            programs = {
+              # Python formatting
+              #ruff-format.enable = true;
+              # TODO: maybe instead of pylint?
+              # ruff-lint.enable = true;
```

Comment:
Most people I know switched to that these days. It's quite fast and has quiet a few more checks. Some really useful one like https://docs.astral.sh/ruff/rules/#flake8-bugbear-b